application/src/main/kotlin/net/corda/v5/application/messaging/FlowSession.kt

```diff
@@ -31,6 +37,26 @@ interface FlowSession {
      */
     val counterparty: MemberX500Name
 
```

Comment:
As discussed, the concept of creating a session, then being able to mutate the context, but only until the point the initiated flow is created (first send or receive) is confusing. 
Effectively the context on a session is an immutable snapshot. If the user wants to set context properties for a given session they should only be able to do this at the point of creating the session, the API should be updated to reflect this and the contextProperties on the FlowSession should be read-only